### PR TITLE
feat: sla-7 ensure applied_sla uniqueness

### PR DESCRIPTION
**Pull request by vishnu-narayanan**
# Pull Request Template

## Description

- Ensure `applied_sla` is unique for a conversation-account-sla_policy combination
- Modify action_service to handle the string `sla_policy id `from the automation listener
- Handle the case if a conversation already has an SLA attached
- Improve error handling if SLA is not found, `nil` or `empty`

Fixes https://linear.app/chatwoot/issue/CW-3094/sla-improvements

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- added specs

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: enterprise/app/models/applied_sla.rb [hunk 1]
```diff
@@ -21,5 +21,7 @@ class AppliedSla < ApplicationRecord
   belongs_to :sla_policy
   belongs_to :conversation
 
+  validates :account_id, uniqueness: { scope: %i[sla_policy_id conversation_id] }
```
Comment (sojan-official) [inline enterprise/app/models/applied_sla.rb:24]: We should add a migration to the back this with unique index

File: enterprise/app/services/enterprise/action_service.rb [hunk 2]
```diff
@@ -1,10 +1,15 @@
 module Enterprise::ActionService
-  def add_sla(sla_policy)
+  def add_sla(sla_policy_id)
+    sla_policy = @account.sla_policies.find(sla_policy_id.first)
+    Rails.logger.info "SLA:: Adding SLA #{sla_policy.id} to conversation: #{@conversation.id}"
     @conversation.update!(sla_policy_id: sla_policy.id)
```
Comment (sojan-official) [inline enterprise/app/services/enterprise/action_service.rb:5]: return if @conversation.sla_policy.present ?

File: enterprise/app/services/enterprise/action_service.rb [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
     create_applied_sla(sla_policy)
   end
 
   def create_applied_sla(sla_policy)
+    return if AppliedSla.exists?(conversation_id: @conversation.id, account_id: @conversation.account_id, sla_policy_id: sla_policy.id)
```
Comment (sojan-official) [inline enterprise/app/services/enterprise/action_service.rb:10]: we won't need this check if conversation.sla_policy? is present in the above check right ?

Comment (vishnu-narayanan) [reply enterprise/app/services/enterprise/action_service.rb:10]: True. Will make the change.

File: enterprise/app/services/enterprise/action_service.rb (hunk 2 above)
Comment (vishnu-narayanan) [reply enterprise/app/services/enterprise/action_service.rb:5]: ✅

File: enterprise/app/models/applied_sla.rb (hunk 1 above)
Comment (vishnu-narayanan) [reply enterprise/app/models/applied_sla.rb:24]: Noted

File: spec/enterprise/services/enterprise/action_service_spec.rb [hunk 4]
```diff
@@ -8,16 +8,42 @@
...
+        action_service.add_sla(nil)
+        expect(conversation.reload.sla_policy_id).to be_nil
+      end
+    end
+
+    context 'when conversation already has a sla policy' do
+      it 'does not add the new sla policy to the conversation' do
+        existing_sla_policy = sla_policy
+        new_sla_policy = create(:sla_policy, account: account)
+        conversation.update!(sla_policy_id: sla_policy.id)
```
Comment (sojan-official) [inline spec/enterprise/services/enterprise/action_service_spec.rb:36]:
```suggestion
        conversation.update!(sla_policy_id: existing_sla_policy.id)
```

**Review by sojan-official — COMMENTED** [review]
LGTM

**Review by sojan-official — APPROVED** [review]

**Pull request merged**